### ISSUE TITLE
`autocomplete` attribute for select element (Success Criterion 1.3.5 "Identify Input Purpose")

### ISSUE THREAD

**Issue by caztcha** [w3c/wcag#4162]
This is just a question regarding the interpretation of WCAG Success Criterion "1.3.5 Identify Input Purpose".

If the prefecture input is implemented using `<select>` element, instead of `<input type="text">`, for the user's postal address, is the lack of `autocomplete` attribute is considered a failure of this success criterion? Or should we implement the `<select>` like this?

```
<label for="prefs">Prefecture</label>

<select autocomplete="address-level1" id="prefs">
<option>Hokkaido</option>
:
<option>Tokyo</option>
:
<option>Aichi</option>
:
<option>Kyoto</option>
<option>Osaka</option>
:
<option>Fukuoka</option>
:
<option>Okinawa</option>
</select>
```

According to HTML Living Standard, most of the "autofill field name" including "address-level" seems to be appropriate only for TEXT control, thus inappropriate for `<select>` element. 
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-field

Based on this, I would think that the lack of `autocomplete` in the `<select>` element would not be considered a failure of this success criterion. Am I correct?

Or is it necessary to include `autocomplete` attribute in such `<select>` element, in the hope for future that assistive technologies will use the `autocomplete` value as a cue to present a symbol for users?

I would appreciate it if you could let me know your views on this.

**Comment by TestPartners** [comment]
SC 1.3.5 is not concerned with autofill. The intention is that user agents can identify the purpose of an input and display an appropriate icon or some other means of identification in addition to the text label. As such, it’s ok to use the “autocomplete” attribute on the <select> element. It is explicitly permitted in the HTML5 specification.
https://html.spec.whatwg.org/multipage/form-elements.html#the-select-element

Steve Green
Managing Director
Test Partners Ltd

From: Kazuhiko Tsuchiya ***@***.***>
Sent: 06 December 2024 05:47
To: w3c/wcag ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [w3c/wcag] `autocomplete` attribute for select element (Success Criterion 1.3.5 "Identify Input Purpose") (Issue #4162)


This is just a question regarding the interpretation of WCAG Success Criterion "1.3.5 Identify Input Purpose".

If the prefecture input is implemented using <select> element, instead of <input type="text">, for the user's postal address, is the lack of autocomplete attribute is considered a failure of this success criterion? Or should we implement the <select> like this?

<label for="prefs">Prefecture</label>



<select autocomplete="address-level1" id="prefs">

<option>Hokkaido</option>

:

<option>Tokyo</option>

:

<option>Aichi</option>

:

<option>Kyoto</option>

<option>Osaka</option>

:

<option>Fukuoka</option>

:

<option>Okinawa</option>

</select>

According to HTML Living Standard, most of the "autofill field name" including "address-level" seems to be appropriate only for TEXT control, thus inappropriate for <select> element.
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-field

Based on this, I would think that the lack of autocomplete in the <select> element would not be considered a failure of this success criterion. Am I correct?

Or is it necessary to include autocomplete attribute in such <select> element, in the hope for future that assistive technologies will use the autocomplete value as a cue to present a symbol for users?

I would appreciate it if you could let me know your views on this.

**Comment by patrickhlauke** [comment]
1.3.5 is a bit ... vague (nothing new when it comes to WCAG) here. in theory, `autocomplete` isn't the *only* way in which to pass ... there had been talk of other ways to do things, like RDF attributes or really anything that would count as programmatic information that can convey the purpose of an input... but for better or worse, `autocomplete` is the de-facto way to achieve things in HTML. so, even though there may not be any actual practical use (in terms of ancilliary benefits like actually having browsers autofill things), I'd say yes the absence of an appropriate `autocomplete` would be a failure.

**Comment by JAWS-test** [comment]
> are appropriate only for TEXT input.

in the table it says "TEXT" (not "TEXT input") and this refers to:

https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#control-group-text

and there you find: select

**Comment by caztcha** [comment]
Thank you @JAWS-test 

> in the table it says "TEXT" (not "TEXT input") and this refers to:
> 
> https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#control-group-text
> 
> and there you find: select

Oh my goodness. I overlooked it completely. Thank you so much for pointing it out.

**Comment by mbgower** [comment]
**Draft working group response**
The `autocomplete` attributes apply to `select` elements. In the information following the table in the [hmlt5 spec](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#control-group-text), `select` elements are explicitly listed under each of the group categories in the table.

The working group will add some information to the understanding document or the existing technique (or both) to clarify this.

**Comment by caztcha** [comment]
Thank you for the draft working group response. I appreciate everyone taking time for my question.

**Comment by patrickhlauke** [comment]
To avoid similar confusion, we could do with updating the understanding document to clarify that "input field" doesn't exclusively mean `<input>` elements

**Comment by bruce-usab** [comment]
Discussed on Backlog call 1/17, @patrickhlauke offered to draft PR.  Moved to In progress.
